packages/frontend-shared/src/graphql/urqlClient.ts

```diff
@@ -75,13 +62,16 @@ interface AppUrqlClientConfig {
 
 export type UrqlClientConfig = LaunchpadUrqlClientConfig | AppUrqlClientConfig
 
-export function makeUrqlClient (config: UrqlClientConfig): Client {
+export async function makeUrqlClient (config: UrqlClientConfig): Promise<Client> {
   let hasError = false
 
+  const dfd = pDefer()
   const exchanges: Exchange[] = [dedupExchange]
 
   const io = window.ws ?? getPubSubSource(config)
 
+  io.on('connect', dfd.resolve)
```

Comment:
Rather than doing a pDefer,  could we say:

```suggestion
  const connectPromise = new Promise((resolve, reject) => { io.on('connect', resolve) })
```

And then await that promise on line 148.

Reply:
Updated!